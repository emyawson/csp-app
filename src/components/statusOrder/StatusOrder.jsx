// Modules
import './StatusOrder.scss';
import PropTypes from 'prop-types';
import React, { Fragment } from 'react';
// Button
import '@ingka/svg-icon/style.scss';
import '@ingka/button/style.scss';
import '@ingka/focus/style.scss';
import Button from '@ingka/button';
import RichTextEditor from '../RichTextEditor/RichText.jsx';
import RecommendedLabel from '../recommendedLabel/RecommendedLabel.jsx';

const StatusOrder = ({
  heading, content, label, urlButton,
}) => (
    <Fragment>
      <div className="status-order">
        <RecommendedLabel />
        <h2 className="status-order__heading">{heading}</h2>

        {content && <RichTextEditor data={content} style={'status-order__description'}/>}
        {/* the URL is dummy for now */}
        <Button type="primary" className="status-order__btn" onClick={() => { window.location.href = urlButton; }}>{label}</Button>
        </div>
    </Fragment>
);

StatusOrder.propTypes = {
  heading: PropTypes.string,
  content: PropTypes.object,
  label: PropTypes.string,
  urlButton: PropTypes.string,
};

export default StatusOrder;
