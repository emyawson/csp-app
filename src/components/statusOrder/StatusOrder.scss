@import '../../index';

.status-order {
  @include respond(tablet) {
    &__description {
      min-width: 334px;
    }

    &__heading {
      max-width: 293px;
    }
  }
  @include respond(mobile) {
    &__description {
      min-width: unset;
      width: 100%;
    }

    &__heading {
      width: 100%;
    }
  }

  &__description {
    font-size: $font-S;
    margin-top: 16px;
    max-width: 607px;
  }

  &__heading {
    @extend %large-titlebase;
    max-width: 293px;
  }

  &__btn {
    margin-top: 24px;
  }
};
