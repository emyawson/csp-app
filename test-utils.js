import React from 'react';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import { Router } from 'react-router-dom';
import { connectRouter } from 'connected-react-router';
import { render } from '@testing-library/react';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import { history } from './src/store';
import reducers from './src/store/reducer/index';

const createRootReducer = (reducerHistory) => combineReducers({
  router: connectRouter(reducerHistory),
  ...reducers,
});

// Setup redux persist
const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
};
const persistedReducer = persistReducer(
  persistConfig,
  createRootReducer(history),
);

const renderWithRouter = (ui, options = {}, initialState) => ({
  ...render(
      <Provider store={createStore(persistedReducer, initialState)}>
        <Router history={history}>{ui}</Router>
      </Provider>,
  ),
});

const AllTheProviders = () => {};
const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, renderWithRouter };
