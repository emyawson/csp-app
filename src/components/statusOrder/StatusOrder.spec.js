import React from 'react';
// import { render, screen } from "@testing-library/react";
import StatusOrder from './StatusOrder.jsx';
// import { renderWithRouter } from "../test-utils";
import { renderWithRouter } from '../../../test-utils';

const defaultState = {};

const renderComponent = (state = {}) => {
  const combinedState = {
    ...defaultState,
    ...state,
  };
  return renderWithRouter(<StatusOrder />, combinedState);
};

test('renders <StatusOrder/>', () => {
  const { container } = renderComponent();
  expect(container).toMatchSnapshot();

  // const linkElement = screen.getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();
});
