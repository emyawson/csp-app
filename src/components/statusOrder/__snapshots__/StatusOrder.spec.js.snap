// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <StatusOrder/> 1`] = `
<div>
  <div
    class="status-order"
  >
    <p
      class="label"
    >
      Aanbevolen
    </p>
    <h2
      class="status-order__heading"
    />
    <button
      class="btn btn--primary status-order__btn"
      type="button"
    >
      <span
        class="btn__inner"
      >
        
      </span>
    </button>
  </div>
</div>
`;
